docs/fundamentals/code-analysis/configuration-files.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Configuration files for code analysis rules
+description: Learn about different configuration files to configure code analysis rules.
+ms.date: 09/24/2020
+ms.topic: conceptual
+no-loc: ["EditorConfig"]
+---
+# Configuration files for code analysis rules
+
+Code analysis rules have various [configuration options](configuration-options.md). These options are specified as key-value pairs in an analyzer configuration file. Following configuration files can be used to specify these options:
+
+- [EditorConfig](#editorconfig) file: File or folder based configuration options
+- [Global AnalyzerConfig](#global-analyzerconfig) file: Project level configuration options
+
+## EditorConfig
+
+[EditorConfig](/visualstudio/ide/create-portable-custom-editor-options) files are used to provide **options that apply to specific source files or folders**. Options are placed under section headers to identify the applicable files and folders. Add an entry for each rule you want to configure, and place it under the corresponding file extension section, for example, `[*.cs]`.
+
+```ini
+[*.cs]
+<%option_name%> = <%option_value%>
+```
+
+In the above example, `[*.cs]` is an editorconfig section header to select all C# files with `.cs` file extension within the current folder, including subfolders. The subsequent entry `<%option_name%> = <%option_value%>` is an analyzer option that will be applied to all these C# files.
+
+You can apply EditorConfig file conventions to a folder, a project, or an entire repo by placing the file in the corresponding directory. These options will be applied when executing the analysis at build time as well as inside Visual Studio.
+
+If you have an existing *.editorconfig* file for editor settings such as indent size or whether to trim trailing whitespace, you can place your code analysis configuration options in the same file.
+
+> [!TIP]
+> Visual Studio provides an *.editorconfig* item template that makes is easy to add one of these files to your project. For more information, see [Add an EditorConfig file to a project](/visualstudio/ide/create-portable-custom-editor-options#add-an-editorconfig-file-to-a-project).
+
+### Example
+Following is an example EditorConfig file to configure options and rule severity:
+
+```ini
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+#### .NET Coding Conventions ####
+
+# this. and Me. preferences
+dotnet_style_qualification_for_method = true:warning
+
+#### Diagnostic configuration ####
+
+# CA1000: Do not declare static members on generic types
+dotnet_diagnostic.CA1000.severity = warning
+```
+
+## Global AnalyzerConfig
```

Comment:
Should we be using the term "AnalyzerConfig" or is that just an internal term? Would this be better?
```suggestion
## Global configuration files
```

Reply:
I have used `Global AnalyzerConfig` terminology elsewhere, so we should probably retain this for consistency.